packages/server/routes/api/guilds.js

```diff
@@ -239,7 +243,244 @@ router.get(
+        status: "error",
+        message: error.message,
+      });
+    }
+  },
+);
+
+// Update guild member's role
+router.put(
+  "/:guildId/members/:userId/roles",
+  AuthController.authenticate,
+  guildIdValidator,
+  userIdValidator,
+  updateGuildMemberRoleValidator,
+  async (request, response) => {
+    const result = validationResult(request);
+
+    if (!result.isEmpty()) {
+      return response.status(400).json({
+        status: "fail",
+        data: result.array(),
+      });
+    }
+
+    const data = matchedData(request);
+    const guildId = data.guildId;
+    const userId = data.userId;
+    const role = request.body.role;
+
+    try {
+      const client = request.user;
+      const clientData = await GuildController.getGuildMember(
+        guildId,
+        client.userId,
+      );
+
+      if (!(await GuildController.isGuildMember(guildId, userId))) {
+        return response.status(403).json({
+          status: "fail",
+          data: {
+            userId: "Guild member does not exist.",
+          },
+        });
+      }
+
+      if (!clientData || clientData.role !== GuildMemberRole.Owner) {
+        return response.status(403).json({
+          status: "fail",
+          data: {
+            userId: "Access denied. Only the guild owner can edit roles.",
+          },
+        });
+      }
+
+      if (client.userId === userId) {
+        return response.status(403).json({
+          status: "fail",
+          data: {
+            userId: "The guild owner cannot edit their own role.",
+          },
+        });
+      }
+
+      if (role === GuildMemberRole.Owner) {
+        await GuildController.updateGuildMemberRole(
+          guildId,
+          client.userId,
+          GuildMemberRole.Member,
+        );
+      }
+
+      return response.status(200).json({
+        status: "success",
+        data: {
+          updatedMember: await GuildController.updateGuildMemberRole(
+            guildId,
+            userId,
+            role,
+          ),
+        },
+      });
+    } catch (error) {
+      return response.status(500).json({
+        status: "error",
+        message: error.message,
+      });
+    }
+  },
+);
+
+// Delete guild member
+router.delete(
+  "/:guildId/members/:userId",
+  AuthController.authenticate,
+  guildIdValidator,
+  userIdValidator,
+  async (request, response) => {
+    const result = validationResult(request);
+
+    if (!result.isEmpty()) {
+      return response.status(400).json({
+        status: "fail",
+        data: result.array(),
+      });
+    }
+
+    const data = matchedData(request);
+    const guildId = data.guildId;
+    const userId = data.userId;
+
+    try {
+      // If member does not exist, fail
+      if (!(await GuildController.isGuildMember(guildId, userId))) {
+        return response.status(403).json({
+          status: "fail",
+          data: {
+            userId: "Guild member does not exist.",
+          },
+        });
+      }
```

Comment:
Same thing in my other comment about moving this validation logic to an `express-validator` instead.